src/engine/strat_engine/engine.rs

```diff
@@ -122,6 +153,40 @@ impl Engine for StratEngine {
     fn pools(&self) -> Vec<&Pool> {
         self.pools.into_iter().map(|x| x as &Pool).collect()
     }
+
+    fn get_eventable(&mut self) -> EngineResult<Option<Box<Eventable>>> {
+        Ok(Some(Box::new(DM::new()?)))
+    }
+
+    fn evented(&mut self) -> EngineResult<()> {
+        let device_list = DM::new()?.list_devices()?;
+
+        for &(ref dm_name, _, new_event_nr) in &device_list {
+            let new_event_nr = new_event_nr.expect("Supported DM versions always provide a value");
+            if let Some(prev_event_nr) = self.watched_dev_last_event_nrs.get(dm_name) {
+                if new_event_nr > *prev_event_nr {
+                    // Event! We know the dm name, but don't know which
+                    // pool
+                    for pool in &mut self.pools {
+                        if pool.get_eventing_dev_names().contains(dm_name) {
+                            pool.event_on(dm_name)?;
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        self.watched_dev_last_event_nrs = device_list
+            .into_iter()
+            .filter_map(|(dm_name, _, event_nr)| match event_nr {
```

Comment:
There's no need to filter here anymore, since ```even_nr``` can not be ```None```.

Reply:
ok